src/devices/rolling_code.c

```diff
@@ -0,0 +1,335 @@
+/** @file
+    Decoder for Rolling Code Transmitter
+
+    Copyright (C) 2020 David E. Tiller
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+ */
+
+/**
+ The device uses OOK_PULSE_PCM_RZ encoding.
+ The packet starts with either a narrow (500 uS) start pulse or a long (1500 uS) pulse.
+ - 0 is defined as a 1500 uS gap followed by a  500 uS pulse.
+ - 1 is defined as a 1000 uS gap followed by a 1000 uS pulse.
+ - 2 is defined as a  500 uS gap followed by a 1500 uS pulse.
+
+ Transmissions consist of a '1' length packet of 20 'trits' (trinary digits)
+ followed by a '3' length packet of 20 trits. These two packets are repeated
+ some number of times.
+ The trits represent a rolling code that changes on each keypress, a fixed
+ 16 trit device ID,  3 id trits (key pressed), and a 1 trit button id.
+ All of the data is obfuscated and a 1 length and a 3 length packet are
```

Comment:
2
All of the data is obfuscated and a 1 length and a 3 length packet are
￼ required to successfully decode a transmission.